src/lib/components/TraceUpdater.react.js

```diff
@@ -17,96 +14,79 @@ let _prev_layout = null;
-                for (let i = 1; i < updateData.length; i++) {
-                    trace = updateData[i];
-                    // get the trace it's index and delete it from the trace object
-                    // as it is not a valid attribute for Plotly.restyle
-                    index = trace.index
-                    delete trace.index;
-
-                    if (Array.isArray(trace.x) && trace.x.length < 1) {
-                        delete trace.x;
-                        delete trace.y;
-                    }
-
-                    if (trace != null && index != null) {
-                        // put everything in the right format & call restyle for each
-                        // trace
-                        for (const colName of traceColNames) {
-                            if (trace[colName] == null) {
-                                delete trace[colName];
-                            }
-                            else if (Array.isArray(trace[colName])) {
-                                trace[colName] = [trace[colName]];
-                            }
-                        }
-                        Plotly.restyle(graphDiv, trace, index);
-                    }
-                }
-            }
-            else {
-                // Create a set-union of all the to-be-updated traces their 
-                // first-level keys
-                s_keys = new Set();
-                for (let i = 1; i < updateData.length; i++) {
-                    Object.keys(updateData[i]).forEach(key => s_keys.add(key));
-                }
-
-                // new variable to store the updated data in a compatible format to 
-                // call restyle only once
-                const mergedUpdateData = { };
-                for (const k of s_keys) {
-                    mergedUpdateData[k] = [];
-                }
-                for (let i = 1; i < updateData.length; i++) {
-                    trace = updateData[i];
-                    // delete the x & y trace if the length of the array is < 1
-                    if (Array.isArray(trace.x) && trace.x.length < 1) {
-                        delete trace.x;
-                        delete trace.y;
-                    }
-
-                    for (const k of s_keys) {
-                        if (trace[k] === null) {
-                            mergedUpdateData[k].push([]);
-                        } else {
-                            mergedUpdateData[k].push(trace[k]);
-                        }
-                    }
-                }
-                let index_arr = mergedUpdateData['index'];
-                delete mergedUpdateData['index'];
-                Plotly.restyle(graphDiv, mergedUpdateData, index_arr);
-            }
+        /// VALIDATION ///
+
+        const { id, gdID, sequentialUpdate, updateData } = this.props;
+        if (!isArray(updateData) || !head(updateData.length)) throw Error(`Invalid updateData '${updateData}'`);
```

Comment:
What is the expected behaviour of calling `head` on `updatedata`.**`length`**; 

I assume you intend to call `!head(updateData)`

Reply:
Ah yes, that’s a mistake.